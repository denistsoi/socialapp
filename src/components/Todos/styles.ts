import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  todo: {
    flex: 1,
    flexDirection: "row"
  },
  todoText: {
    width: "90%",
    color: "#666"
  },
  iconColor: {
    color: "#666"
  }
});

export default styles;
