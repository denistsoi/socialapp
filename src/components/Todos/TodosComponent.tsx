import React, { ReactElement } from "react";
import { View, Text } from "react-native";
import Icon from "react-native-vector-icons/MaterialIcons";

import { Todo } from "../../types/resource";

import styles from "./styles";

const Todos = ({ todos }: { todos: Array<Todo> }): ReactElement => {
  return (
    <View>
      {todos?.map((todo, index) => (
        <View key={index} style={styles.todo}>
          <Icon
            size={20}
            style={styles.iconColor}
            name={
              todo.completed ? "check-box" : "check-box-outline-blank"
            }></Icon>
          <Text style={styles.todoText} numberOfLines={1}>
            {todo.title}
          </Text>
        </View>
      ))}
    </View>
  );
};

export default Todos;
