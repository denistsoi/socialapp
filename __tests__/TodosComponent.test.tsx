import React from "react";
import { render } from "react-native-testing-library";
import Todos from "../src/components/Todos/TodosComponent";

describe("Todos Component", () => {
  it.only("should render single data in todos", async () => {
    let props: any;

    props = {
      onPress: jest.fn(),
      todos: [
        {
          id: 1,
          userId: 1,
          title: "Buy Groceries",
          completed: false
        }
      ]
    };

    const component = render(<Todos todos={props.todos} />);

    expect(component.toJSON()).toMatchSnapshot();
    expect(component.getByText("Buy Groceries")).toBeDefined();
  });
});
