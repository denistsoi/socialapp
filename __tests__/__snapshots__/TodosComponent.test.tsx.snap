// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos Component should render single data in todos 1`] = `
<View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    Icon
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#666",
          "width": "90%",
        }
      }
    >
      Buy Groceries
    </Text>
  </View>
</View>
`;
